ASIM/dev/ASimTester/filteringTest/filteringTest.py

```diff
@@ -0,0 +1,227 @@
+from azure.core.exceptions import HttpResponseError
+from azure.identity import DefaultAzureCredential
+from schemasParameters import all_schemas_parameters
+
+
+DUMMY_VALUE = "\'!not_REAL_vAlUe\'"
+
+
+argparse_parser = argparse.ArgumentParser()
+argparse_parser.add_argument("ws_id", help = "workspace ID")
+argparse_parser.add_argument("parser_file_relative_path", help = "relative path to the parser file")
+argparse_parser.add_argument("days_delta", default = 1, type = int, help = "number of days to query over ,default is set to  1", nargs='?')
+args = argparse_parser.parse_args()
+ws_id = args.ws_id
+parser_file_path = args.parser_file_relative_path
+days_delta = args.days_delta
+
+
+end_time = datetime.now(timezone.utc)
+start_time = end_time - timedelta(days = days_delta)
+
+
+def attempt_to_connect(credential):
+    try:
+        with contextlib.redirect_stderr(None):
+            client = LogsQueryClient(credential)
+            empty_query = ""
+            response = client.query_workspace(
+                    workspace_id = ws_id, 
+                    query = empty_query,
+                    timespan = timedelta(days = 1)
+                    )
+            if response.status == LogsQueryStatus.PARTIAL or response.status == LogsQueryStatus.FAILURE:
+                raise Exception()
+            else:
+                return client
+    except Exception as e:
+        return None
+    
+
+# Authenticating the user
+client = attempt_to_connect(DefaultAzureCredential(exclude_interactive_browser_credential = False))
+if client is None:
+    client = attempt_to_connect(DefaultAzureCredential(exclude_interactive_browser_credential = False, exclude_shared_token_cache_credential = True))
+    if client is None:
+        print("Couldn't connect to workspace")
+        sys.exit()
+
+
+def get_parser(parser_path):
+    try:
+        with open(parser_path, 'r') as file_stream:
+            return yaml.safe_load(file_stream)
+    except:
+        raise Exception()
+
+
+# Creating a string of the parameters of a parser
+def create_parameters_string(parser_file):
+    paramsList = []
+    for param in parser_file['ParserParams']:
+        paramDefault = f"\'{param['Default']}\'" if param['Type']=="string" else param['Default']
+        paramsList.append(f"{param['Name']}:{param['Type']}={paramDefault}")
+    return ','.join(paramsList)
+    
+
+def create_query_without_call(parser_file):
+    params_str = create_parameters_string(parser_file) 
+    query_from_yaml = parser_file['ParserQuery']
+    return f"let query= ({params_str}) {{ {query_from_yaml} }};\n"
+
+
+def create_call_without_parameter(column_name):
+    return f"query() | summarize count() by {column_name}\n" 
+
+
+def create_call_with_parameter(parameter, value, column_name):
+    return f"query({parameter}={value}) | summarize count() by {column_name}\n"
+
+
+class FilteringTest(unittest.TestCase):
+    # "Main" function which runs initiates the validations 
```

Comment:
>Main" function which runs initiates the validations [](http://example.com/codeflow?start=7&length=51)

Not very clear description, please rephrase

Reply:
Added more description about what the function does.